src/mvs_eland/C0_data_processing.py

```diff
@@ -1587,22 +1591,21 @@ def receive_timeseries_from_csv(
             }
         )
 
-        if dict_asset[OPTIMIZE_CAP][VALUE] is True:
-            logging.debug("Normalizing timeseries of %s.", dict_asset[LABEL])
-            dict_asset.update(
-                {
-                    TIMESERIES_NORMALIZED: dict_asset[TIMESERIES]
-                    / dict_asset[TIMESERIES_PEAK][VALUE]
-                }
+        logging.debug("Normalizing timeseries of %s.", dict_asset[LABEL])
+        dict_asset.update(
+            {
+                TIMESERIES_NORMALIZED: dict_asset[TIMESERIES]
+                / dict_asset[TIMESERIES_PEAK][VALUE]
+            }
+        )
+        # just to be sure!
+        if any(dict_asset[TIMESERIES_NORMALIZED].values) > 1:
+            logging.warning(
+                "Error, %s timeseries not normalized, greater than 1.",
+                dict_asset[LABEL],
```

Comment:
Puh, this section with "input" and so on is really, really messy. I see what you are doing and it seems right, I hope that our benchmark tests protect us from any weird consequences...

Have you checked the consequence of always adding parameter `TIMESERIES_NORMALIZED` to the `energyProvider` assets? An oemof asset is added demending on whether or not his parameter is included [here](https://github.com/rl-institut/mvs_eland/blob/dev/src/mvs_eland/D1_model_components.py#L687) and [here](https://github.com/rl-institut/mvs_eland/blob/dev/src/mvs_eland/D1_model_components.py#L745) in `D1`. Either we skip an important function with this or we can delete some code there.

Reply:
I guess this part had already been checked. I only deleted one line (`if dict_asset[OPTIMIZE_CAP][VALUE] is True:`. The whole block is green because the indentation has changed due to deleting the if statement.

Reply:
> I guess this part had already been checked. 

Where?

> I only deleted one line (`if dict_asset[OPTIMIZE_CAP][VALUE] is True:`. The whole block is green because the indentation has changed due to deleting the if statement.

Yeah, I know.... but this nevertheless might have implications for the code in `D1`. I can also imagine that the code in `D1` is excess code by now: We are by now not putting the availability schedule onto the source of a DSO but onto a transformer instead. I think we need to look at this, if not now than we need to remind ourselves with an issue and create some tests.